### ISSUE TITLE
PDF IFilter broken

### ISSUE THREAD

**Issue by trudnorx** [sumatrapdfreader/sumatrapdf#3372]
It's broken (at least on some OSes?) as of the newest Sumatra version; I am unsure what update broke it. But it is not an indexing issue on my end. Content search is not working.
Tested on Windows 7.

**Comment by GitHubRulesOK** [comment]
mine seems good, so its a broken machine ?
![image](https://user-images.githubusercontent.com/12754913/226605757-5a0ac06d-fdd6-4a8c-bf39-d6db48be6914.png)

**Comment by GitHubRulesOK** [comment]
There should AFAIK be no general changes in that iFilter  for a long time but thats not to say there is not a knnock on from one of the thousands of other changes each month as the underlying functions are 3rd party decoding of pdf contents. 
however I have never noticed any specific glitch on the units where i have it installed and thus switched on.
if it is platform specific per unit I would reset the windows indexing as it sounds like that machines index is not correct, but indexing is a threaded process given lower priority over other windows tasks, so indexing a drive full of PDFs can take ages.
Results can be miliseconds for small folders after those are indexed by Windows Search. 

For a good stable index there are tools such as MythicSoft and others that can use Acrobat or PDF-lib TET or SumatraPDF filters I would suggest TET may be a good alternative

**Comment by trudnorx** [comment]
Still no issue with search or indexing at all, already checked everything like that, it is all correct.
Maybe you can test it on a Windows 7 VM to see if it is an OS issue?

**Comment by GitHubRulesOK** [comment]
I am uncertain if it would work in a Win7 VM I had enough problems using it in Windows own Version11 VM so usually had it switch off there (same as I usually kept it off in 95 98 XP Vista & 7/8) only use it in 10/11 because the hardware is good enough

**Comment by GitHubRulesOK** [comment]
OK that was quicker than anticipated
So added Search to the downloads folder
Typed in dissertation it found the only pdf with that word.

![Untitled](https://user-images.githubusercontent.com/12754913/227519933-8262d608-9043-44b4-b3bc-d15958fb9a67.png)

**Comment by trudnorx** [comment]
Thank you for your help so far.
It seems like it is not dependent on PDF file. It is not finding any word in any PDF file, even files that were working perfectly with the older version. I am unsure if version is even the issue anymore. Or whether it is a problem with my machine or Sumatra...
Though I still doubt it is related to indexing issues since I tried everything...
The only idea I have is that maybe changing the Sumatra version triggered some bad change in the registry or elsewhere (installer bug?) which caused the problems.
However, I checked the registry and the IFilter seems to be registered properly.
So I'm really unsure what's going on.

**Comment by GitHubRulesOK** [comment]
search index can become deleted or faulty 
however this is windows SumatraDF is only a filter thus aids search no control by SumatraPDF

Place some test files in a clean folder (add a sprinkle of plain text files) set the folder to indexed and check the plain text words are found in simple search, note it will not work in drive V: but it can when moved to drive F: (its local not network managed)

There are other areas in hard drive c: that are part of exclusions but you need to decide if the files are excluded from searching

![image](https://user-images.githubusercontent.com/12754913/227742333-a65af086-74a0-4518-92d6-6855e6e63ea8.png)
![image](https://user-images.githubusercontent.com/12754913/227743115-10f969f9-204b-4bef-9d8b-b8c06858a974.png)

ensure sumatrafilter is set BY WINDOWS to contents
![image](https://user-images.githubusercontent.com/12754913/227742537-4494fc47-a2e2-47e3-80b5-1722af5183ce.png)

**Comment by trudnorx** [comment]
Again, thank you for your help. I had already checked everything you mentioned. Except one thing: moving to another drive. It seems after the PDF file was moved to another drive, the searching works correctly there (despite the fact there is no index for this drive). I guess maybe the index is damaged. But I had already rebuilt the index several times (deleting from the Windows UI and starting from zero), so I don't understand how. The search is still not working for PDF files in my original drive, while other file types work... (Any idea?)

Still, what triggered this problem is updating SumatraPDF, which changed to the new IFilter version and somehow broke the functioning index. So maybe there is some bug with the installer, or Windows.

**Comment by GitHubRulesOK** [comment]
SumatraPDF as I keep stating is, AFAIK only an aid to search function.
The Microsoft Windows Search app is responsible for doing and using an index file the only way an ifilter has an influence is the text must be valid plain text optionally from the document or zips contents. 

During un/re/install the support for reading decoded PDFs may be lost and the index will do whatever Microsoft decides in such cases, SumatraPDF does not amend the index. 

On installation the filter is registered for use system wide, so if it works in one folder it can be used system wide, but again ALL that usage per user/location is 100% Windows Search application. So SumatraPDF (Or any iFilter) either works 100% or not works at all, depending on the search application.

However I have to agree its never clear why its not working hence the trouble-shooter is not a good means to see what is failing. Thus I have had variable results in some situaions but its never due to the simple shim between pdf content and search.exe

**Comment by trudnorx** [comment]
Already checked all this too. It seems that the filter is correctly installed but only works for drives that are not C, weird. And it was working for C before updating Sumatra to the new version. It seems neither downgrading nor reinstalling helps, nor fixing the index, nor anything else you have suggested so far...

**Comment by GitHubRulesOK** [comment]
the index exclusions list is the master for what folders are inclusive/exclusive so if root of c:\ is an exclusion nothing on the drive will be included.

**Comment by GH4TP** [comment]
I found that windows search via explorer was not searching inside PDFs and ended up replacing Sumatra iFilter with Adobe's. An index rebuild with Sumatra's iFilter took substantially less time than with the Adobe one - I assume because, with the Sumatra iFilter, PDF content wasn't being indexed.

**Comment by GitHubRulesOK** [comment]
@GH4TP 

The topic is variable so for example I use SumatraPDF knowing it cannot find as much entries as Adobe nor TeT (the other one that I recommend)

If I search in current folder and sub folders for a named file it will take more than 40 seconds to scan all the filenames and return that there are 3439 PDFs 
Now a large number would have no text to index and others are protected against searching so if I search for "THE" I am not surprised that in 2 seconds explorer says 1050 (E on its own is only 1595) 

I would expect Adobe to be slower building its index because it will include more contents than SumatraPDF, but usually I know the keyword I want.

So I have been recently working somewhere on a few Texas files and searching for Texas returns 40 instantly but I know better it was a contract so now in 2 seconds I can see 6 from 3439 including the 3 I was looking for.

**Comment by GH4TP** [comment]
> TeT (the other one that I recommend)

Great tip, thank you!

**Comment by kjk** [comment]
I think it was caused by not signing `PdfFilter.dll` and setups configured to not load unsigned dlls.

I've started signing the dll so it should be fixed now.

If not, please open a new bug (this one has too many comments already)

**Comment by Neustradamus** [comment]
@kjk: Any progress on this problem?

**Comment by Neustradamus** [comment]
@kjk: Latest good is 3.1.2, all recents do not work correctly.
Can you fix it?

**Comment by kjk** [comment]
@Neustradamus ifilter in latest pre-release works for me.

if it doesn't for you then I would need a lot of information about your setup and how you're verifying it doesn't work

**Comment by Neustradamus** [comment]
@kjk: I have tested all versions on Windows 7, the latest good one is 3.1.2 (some versions have not "All users" in installer too).

SumatraPDF 3.6.1 can work on Windows 7 except that the .pdf files can not be by default opened with SumatraPDF.

All times, we need to do, manually, "Open with".

**Comment by GitHubRulesOK** [comment]
@Neustradamus 
I use windows 7 32 bit as a server thus the files are accessible across my LAN but that is not important.
The point is the ifilter from current 32 bit prerelease is working in the folders I ask it to.

So there are 18 files in that folder or drive path of type:pdf with the word free (or freedom or whatever starts as free) once opened I can focus on the one word in many that is a match.

<img width="1086" height="661" alt="Image" src="https://github.com/user-attachments/assets/6b163e65-daa1-402f-a470-a6ffa4255984" />

The default file handler is the system Edge but 7 thinks that is Internet Explorer (as it used to be) so to set the default pdf handler I would need to choose one of the many from 

<img width="661" height="661" alt="Image" src="https://github.com/user-attachments/assets/ff7e36a8-e0fe-43b3-8afe-ab4fab7e826e" />

**Comment by kjk** [comment]
## Windows 7 and the IFilter

There is **no intentional "IFilter doesn't work on Windows 7"** path in the code.

- We still target Win7 (`WINVER` / `_WIN32_WINNT` = `0x0601`).
- `PdfFilter.dll` is built with Vista-class PE OS/subsystem (6.0), static CRT, and ordinary Win32 + `libmupdf.dll` imports — nothing that obviously requires Win8+ only for the filter to load.
- Registration is classic COM / `PersistentHandler` (same model Search uses on Win7).
- Independent check: current **32-bit prerelease** IFilter has been reported working on **Windows 7** (see comments above).

### "3.1.2 was the last good" / "always Open with"

Please keep these separate:

| Symptom | What it is |
|--------|------------|
| PDF content search / indexing | **IFilter** (`PdfFilter.dll`) |
| PDFs don't open with Sumatra by default; need **Open with** every time | **File association** — not the IFilter |

IFilter only feeds Windows Search text extraction. It does **not** set the default PDF app. Win7 association UI is different from modern "default apps," so association issues can show up even when the app and filter load fine.

### Real reasons IFilter fails (including on Win7)

1. **Bitness** — SearchIndexer must match the filter (64-bit OS → 64-bit install / `PdfFilter.dll`). Wrong bitness: registered but never used.
2. **`libmupdf.dll` next to the filter** — `PdfFilter.dll` depends on it. If the indexer can't load MuPDF from the install dir, extraction fails.
3. **Signing / code integrity** — unsigned `PdfFilter.dll` can be blocked where policy requires signed DLLs (not Win7-specific; we started signing the DLL for this).
4. **Windows Search setup** — includes/excludes, "Always search file names and contents," folder not indexed, index corruption after reinstall. Reinstall re-registers the filter; Windows does not magically rebuild a correct index for that.
5. **Installer** — Search filter only if that option was checked / filter is registered (HKLM vs HKCU can matter for the Search service).
6. **Content limits** — no OCR for scans; encrypted PDFs without a password in the filter path, etc.

### Bottom line

- **Hard Win7 incompatibility in code?** No.
- **Can IFilter work on Win7?** Yes (and has been confirmed with current prerelease on Win7).
- **If it fails for you:** check 32 vs 64 install, that the Search filter was actually registered, that `PdfFilter.dll` + `libmupdf.dll` load under Search, and don't confuse default PDF app with content search.

If you still see a failure after that, open a **new** issue with: OS bitness, Sumatra bitness/version, whether "Windows Search filter" was installed, and how you're testing (Explorer content search vs filename-only, sample path).

**Comment by Neustradamus** [comment]
@kjk: Sorry, so I need to create the ticket for File association.
